src/java/voldemort/store/readonly/StoreVersionManager.java

```diff
@@ -78,7 +78,7 @@ public void syncInternalStateFromFileSystem() {
         // Make sure versions missing from the file-system are cleaned up from the internal state
         for (Long version: versionToEnabledMap.keySet()) {
             File[] existingVersionDirs = ReadOnlyUtils.getVersionDirs(rootDir, version, version);
-            if (existingVersionDirs.length == 0) {
+            if (existingVersionDirs == null || existingVersionDirs.length == 0) {
```

Comment:
This doesn't feel right to me, ReadOnlyUtils.getVersionDirs should return an empty list. There are too many callers, if there is a semantic change introduced, I would expect it to return an empty list instead of handling it all the callers.